search-index/index_data.py

```diff
@@ -225,16 +225,30 @@ def run_cli(
         index_name=os.environ["OPENSEARCH_INDEX"],
         # TODO: convert to env variables?
         opensearch_connector_kwargs={
-            "use_ssl": False,
-            "verify_certs": False,
-            "ssl_show_warn": False,
+            "use_ssl": os.environ["OPENSEARCH_USE_SSL"],
+            "verify_certs": os.environ["OPENSEARCH_VERIFY_CERTS"],
+            "ssl_show_warn": os.environ["OPENSEARCH_SSL_WARNINGS"],
         },
-        embedding_dim=embedding_dim,
+        embedding_dim=int(os.environ["OPENSEARCH_INDEX_EMBEDDING_DIM"]),
     )
 
     opensearch.delete_and_create_index()
+    # We disable index refreshes during indexing to speed up the indexing process,
+    # and to ensure only 1 segment is created per shard. This also speeds up KNN
+    # queries and aggregations according to the Opensearch and Elasticsearch docs.
+    opensearch.set_index_refresh_interval(-1, timeout=60)
     opensearch.bulk_index(actions=doc_generator)
 
+    # TODO: we wrap this in a try/except block because for now because sometimes it times out
```

Comment:
Is the timeout here just a flaky error that occurs occasionally? I'm ignorant of what the opensearch py method is doing under the hood here, but if it's not pinging the server intelligently according to the timeout interval it might be better to wrap this in a retry loop if it's likely to guarantee success after a few tries, in case we forget to reset the interval (how important is this?). This depends on how the method is working under the hood ofc, so may be redundant comment.

Reply:
Good point. I chose to wrap it in a try/except for now as we don't care what the refresh interval is, as we're not incrementally adding documents. 

If(/when) we move to a DAG-managed indexer I imagine the steps of disabling the index refresh, indexing, and re-enabling the index refresh would be separate processes that we could track and re-run if needed.

Relevant part of the Elasticsearch docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html